artifacts.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/mendersoftware/artifacts/parser"
+	"github.com/mendersoftware/artifacts/writer"
+
+	"github.com/urfave/cli"
+)
+
+func wrieArtifact(c *cli.Context) error {
+	if c.String("update") != "" {
+		fmt.Printf("have data file: %v\n", c.String("update"))
+	}
+	if c.String("device-type") != "" {
+		fmt.Printf("have data file: %v\n", c.String("device-type"))
+	}
+	if c.String("image-id") != "" {
+		fmt.Printf("have data file: %v\n", c.String("image-id"))
+	}
+
+	if c.NArg() != 3 ||
+		len(c.String("device-type")) == 0 || len(c.String("image-id")) == 0 {
+		return errors.New("invalid arguments")
+	}
+	aw := awriter.NewWriter("mender", 1)
+	p := parser.RootfsParser{}
+	aw.Register(&p)
+
+	updateTestDir, _ := ioutil.TempDir("", "update")
+	ioutil.WriteFile(filepath.Join(updateTestDir, "meta-data"), []byte(`{"deviceType": "`+c.Args().Get(1)+`", "imageId": "`+c.Args().Get(2)+`"}`), os.ModePerm)
+
+	return aw.WriteSingle(updateTestDir, c.Args().First(), "rootfs-image", filepath.Join(updateTestDir, "mender.tar.gz"))
```

Comment:
Maybe print the output file path to the to stdout? It's easier to use it in the script if the tools tells you where it stored the file. Otherwise the caller will have to make an assumption that about the name and location of the output artifact.